Segment parsing error for audio stream on SCTE splice points live

### What version of Hls.js are you using?

1.6.3

### What browser (including version) are you using?

Chrome Version 131.0.6778.140 (Official Build) (arm64)

### What OS (including version) are you using?

Mac OS 15.1.1 (24B91)

### Test stream

https://live.streaming.a2d.tv/asset/20821722.isml/.m3u8

### Configuration

```JavaScript
{
  "debug": true,
  "enableWorker": true,
  "lowLatencyMode": true,
  "backBufferLength": 90
}
```

### Additional player setup steps

No additional steps required. This is reproducible in demo player with default settings.

### Checklist

- [x] The issue observed is not already reported by searching on Github under https://github.com/video-dev/hls.js/issues
- [x] The issue occurs in the stable client (latest release) on https://hlsjs.video-dev.org/demo and not just on my page
- [x] The issue occurs in the latest client (main branch) on https://hlsjs-dev.video-dev.org/demo and not just on my page
- [x] The stream has correct Access-Control-Allow-Origin headers (CORS)
- [x] There are no network errors such as 404s in the browser console when trying to play the stream

### Steps to reproduce

1. Play the example stream provided until it hits a splice point (Occurs every 5 mins with return after 120 sec)
2. Observe the stream hanging with the segment parsing error


### Expected behaviour

Plays normally

### What actually happened?

Hls-js fails to parse the audio segment that is right before the splice

This does not happen in 1.6.2, so it is a regression in 1.6.3

### Console output

```shell
[log] > [stream-controller]: PARSED->IDLE
audio-track-controller.ts:421 [log] > [audio-track-controller]: Loading audio-track 0 "English" lang:en group:audio-aacl-128 age 3.8 https://live.streaming.a2d.tv/asset/20821722.isml/20821722-audio_0=128000.m3u8
audio-track-controller.ts:101 [log] > [audio-track-controller]: Audio track 0 "English" lang:en group:audio-aacl-128 loaded [455363750-455364156]
base-playlist-controller.ts:216 [log] > [audio-track-controller]: live playlist 0 REFRESHED 455364156--1
base-playlist-controller.ts:346 [log] > [audio-track-controller]: reload live playlist English in 3711 ms
audio-stream-controller.ts:558 [log] > [audio-stream-controller]: Audio track 0 "English" of "audio-aacl-128" loaded [455363750,455364156][part-455364156--1],duration:1562.773299961853
base-stream-controller.ts:918 [log] > [audio-stream-controller]: Loading audio sn: 455364156 of track 0 (frag:[1559.040-1562.773]) cc: 0 [455363750-455364156], target: 1559.04
base-stream-controller.ts:2116 [log] > [audio-stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:525 [log] > [audio-stream-controller]: Loaded audio sn: 455364156 of track 0
transmuxer-interface.ts:396 [warn] > Duration parsed from mp4 should be greater than zero
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:396Understand this warningAI
base-stream-controller.ts:2116 [log] > [audio-stream-controller]: FRAG_LOADING->PARSING
transmuxer-interface.ts:396 [log] > [transmuxer.ts]: Flushed audio sn: 455364156 of track 0
base-stream-controller.ts:2007 [warn] > [audio-stream-controller]: Could not parse fragment 455364156 audio duration reliably (0)
(anonymous) @ base-stream-controller.ts:2007
updateLevelTiming @ base-stream-controller.ts:1998
_handleTransmuxerFlush @ base-stream-controller.ts:1056
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389Understand this warningAI
base-stream-controller.ts:2046 [warn] > [audio-stream-controller]: Found no media in fragment 455364156 of track 0 resetting transmuxer to fallback to playlist timing
updateLevelTiming @ base-stream-controller.ts:2046
_handleTransmuxerFlush @ base-stream-controller.ts:1056
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389Understand this warningAI
base-stream-controller.ts:2116 [log] > [audio-stream-controller]: PARSING->ERROR
main.js:745 Error event: {type: 'mediaError', details: 'fragParsingError', fatal: false, error: Error: Found no media in fragment 455364156 of track 0 resetting transmuxer to fallback to playlist…, frag: Fragment, …}
(anonymous) @ main.js:745
emit @ index.js:203
emit @ hls.ts:386
trigger @ hls.ts:394
updateLevelTiming @ base-stream-controller.ts:2047
_handleTransmuxerFlush @ base-stream-controller.ts:1056
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389Understand this warningAI
base-stream-controller.ts:2116 [log] > [audio-stream-controller]: ERROR->PARSED
base-stream-controller.ts:2063 [log] > [audio-stream-controller]: Parsed audio sn: 455364156 of track 0 (frag:[1559.040-1562.773])
buffer-controller.ts:1012 [warn] > [buffer-controller]: Fragments must have at least one ElementaryStreamType set. type: audio level: 0 sn: 455364156
onFragParsed @ buffer-controller.ts:1012
emit @ index.js:203
emit @ hls.ts:386
trigger @ hls.ts:394
updateLevelTiming @ base-stream-controller.ts:2068
_handleTransmuxerFlush @ base-stream-controller.ts:1056
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389Understand this warningAI
cd5a1f44-329f-4f2c-aa3e-92ad67c9eeca:1168 [log] > Debug logs enabled for "audio" in hls.js version 1.6.3
buffer-controller.ts:1685 [log] > [buffer-controller]: Blocking operation requested, but no SourceBuffers exist
base-stream-controller.ts:721 [log] > [audio-stream-controller]: Buffered audio sn: 455364156 of track 0 (frag:[1559.040-1562.773] > buffer:[1459.200-1559.040])
base-stream-controller.ts:2116 [log] > [audio-stream-controller]: PARSED->IDLE
```

### Chrome media internals output

```shell

```

--- a/src/utils/mp4-tools.ts
+++ b/src/utils/mp4-tools.ts
@@ -723,68 +723,68 @@ export function getSampleData(
       const sampleCount = readUint32(trun, 4);
       const sampleIndex = trackTimes.sampleCount;
       trackTimes.sampleCount += sampleCount;
-      if (track.type === ElementaryStreamTypes.VIDEO) {
-        const dataOffsetPresent = trun[3] & 0x01;
-        const firstSampleFlagsPresent = trun[3] & 0x04;
-        const sampleDurationPresent = trun[2] & 0x01;
-        const sampleSizePresent = trun[2] & 0x02;
-        const sampleFlagsPresent = trun[2] & 0x04;
-        const sampleCompositionTimeOffsetPresent = trun[2] & 0x08;
-        let offset = 8;
-        let remaining = sampleCount;
-        if (dataOffsetPresent) {
+      // Get duration from samples
+      const dataOffsetPresent = trun[3] & 0x01;
+      const firstSampleFlagsPresent = trun[3] & 0x04;
+      const sampleDurationPresent = trun[2] & 0x01;
+      const sampleSizePresent = trun[2] & 0x02;
+      const sampleFlagsPresent = trun[2] & 0x04;
+      const sampleCompositionTimeOffsetPresent = trun[2] & 0x08;
+      let offset = 8;
+      let remaining = sampleCount;
+      if (dataOffsetPresent) {
+        offset += 4;
+      }
+      if (firstSampleFlagsPresent && sampleCount) {
+        const isNonSyncSample = trun[offset + 1] & 0x01;
+        if (!isNonSyncSample && trackTimes.keyFrameIndex === undefined) {
+          trackTimes.keyFrameIndex = sampleIndex;
+        }
+        offset += 4;
+        if (sampleDurationPresent) {
+          sampleDuration = readUint32(trun, offset);
+          offset += 4;
+        } else {
+          sampleDuration = defaultSampleDuration;
+        }
+        if (sampleSizePresent) {
           offset += 4;
         }
-        if (firstSampleFlagsPresent && sampleCount) {
+        if (sampleCompositionTimeOffsetPresent) {
+          offset += 4;
+        }
+        sampleDTS += sampleDuration;
+        rawDuration += sampleDuration;
+        remaining--;
+      }
+      while (remaining--) {
+        if (sampleDurationPresent) {
+          sampleDuration = readUint32(trun, offset);
+          offset += 4;
+        } else {
+          sampleDuration = defaultSampleDuration;
+        }
+        if (sampleSizePresent) {
+          offset += 4;
+        }
+        if (sampleFlagsPresent) {
           const isNonSyncSample = trun[offset + 1] & 0x01;
-          if (!isNonSyncSample && trackTimes.keyFrameIndex === undefined) {
-            trackTimes.keyFrameIndex = sampleIndex;
+          if (!isNonSyncSample) {
+            if (trackTimes.keyFrameIndex === undefined) {
+              trackTimes.keyFrameIndex =
+                trackTimes.sampleCount - (remaining + 1);
+              trackTimes.keyFrameStart = sampleDTS;
+            }
           }
           offset += 4;
-          if (sampleDurationPresent) {
-            sampleDuration = readUint32(trun, offset);
-            offset += 4;
-          } else {
-            sampleDuration = defaultSampleDuration;
-          }
-          if (sampleSizePresent) {
-            offset += 4;
-          }
-          if (sampleCompositionTimeOffsetPresent) {
-            offset += 4;
-          }
-          sampleDTS += sampleDuration;
-          rawDuration += sampleDuration;
-          remaining--;
         }
-        while (remaining--) {
-          if (sampleDurationPresent) {
-            sampleDuration = readUint32(trun, offset);
-            offset += 4;
-          } else {
-            sampleDuration = defaultSampleDuration;
-          }
-          if (sampleSizePresent) {
-            offset += 4;
-          }
-          if (sampleFlagsPresent) {
-            const isNonSyncSample = trun[offset + 1] & 0x01;
-            if (!isNonSyncSample) {
-              if (trackTimes.keyFrameIndex === undefined) {
-                trackTimes.keyFrameIndex =
-                  trackTimes.sampleCount - (remaining + 1);
-                trackTimes.keyFrameStart = sampleDTS;
-              }
-            }
-            offset += 4;
-          }
-          if (sampleCompositionTimeOffsetPresent) {
-            offset += 4;
-          }
-          sampleDTS += sampleDuration;
-          rawDuration += sampleDuration;
+        if (sampleCompositionTimeOffsetPresent) {
+          offset += 4;
         }
-      } else {
+        sampleDTS += sampleDuration;
+        rawDuration += sampleDuration;
+      }
+      if (!rawDuration && defaultSampleDuration) {
         rawDuration += defaultSampleDuration * sampleCount;
       }
     }

